app/src/main/java/ch/epfl/sweng/swenggolf/preference/ListPreferenceAdapter.java

```diff
@@ -9,74 +10,43 @@
-    private ArrayList<User> mDataset;
-
-    public static final User[] USERS_INITIAL = {
-            new User("Anna", "0", "anna@mail.com", "Tomatoes"),
-            new User("Bob", "1", "bob@mail.com", "Screwdriver"),
-            new User("Geany", "2", "geany@mail.com", "Comics"),
-            new User("Greg", "3", "greg@gmail.com", "Ropes"),
-            new User("Fred", "4", "fred@gmail.com", "Beverages"),
-            new User("AAnna", "0", "aanna@mail.com", "Friends"),
-            new User("ABob", "1", "abob@mail.com", "Washing machine"),
-            new User("AGeany", "2", "ageany@mail.com", "Hammer"),
-            new User("AGreg", "3", "agreg@gmail.com", "Lunch"),
-            new User("AFred", "4", "afred@gmail.com", "Cheeseburgers"),
-            new User("BAnna", "0", "banna@mail.com", "Champaign"),
-            new User("BBob", "1", "bbob@mail.com", "Mushrooms"),
-            new User("BGeany", "2", "bgeany@mail.com", "Nothing"),
-            new User("BGreg", "3", "bgreg@gmail.com", "Fries"),
-            new User("BFreEricisSIstirusiwssjdsidjsidskdisjdijsmdisjd",
-                    "4", "fr@gmail.com", "A nice sweatshirt, some hot shoes and a poncho")
-    };
+    private List<User> mDataset = new ArrayList<>();
 
     /**
      * Create a new adapter for the list that fetches information about users.
      * If debug is set to true, a default list of users is used.
      */
     public ListPreferenceAdapter() {
-        if (!debug) {
-            ValueEventListener getUserList = new ValueEventListener() {
-                @Override
-                public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-                    Iterable<DataSnapshot> usersData = dataSnapshot.getChildren();
-                    ArrayList<User> users = new ArrayList<>();
-                    for (DataSnapshot user : usersData) {
-                        users.add(Config.getUser());
-                    }
-                    mDataset = users;
-                }
+        Database d = Database.getInstance();
+        ValueListener<List<User>> getUserList = new ValueListener<List<User>>() {
+            @Override
+            public void onDataChange(List<User> value) {
+                mDataset = value;
+                ListPreferenceAdapter.this.notifyDataSetChanged();
+            }
 
-                @Override
-                public void onCancelled(@NonNull DatabaseError databaseError) {
-                    Log.d("Loading users error", "failed to retrieve users list for preferences");
-                }
-            };
-            DatabaseReference ref = FirebaseDatabase.getInstance().getReference("users");
-            ref.addValueEventListener(getUserList);
-        } else {
-            mDataset = new ArrayList<User>(Arrays.asList(USERS_INITIAL));
-        }
+            @Override
+            public void onCancelled(DbError error) {
+                Log.d("DBError", "Failed to load users");
```

Comment:
DbError is an enum that has a lot of details about errors, e.g. `OPERATION_FAILED`, `PERMISSION_DENIED`, ...
It could be a good idea to use it in the error message (if possible, since it is our custom enum, it might not be printable). Another thing is the tag, it can be find in the documentation that it is 
>Used to identify the source of a log message. It usually identifies the class or activity where the log call occurs.
But again, I might be nitpicking, and we already did otherwise in the project.